### ISSUE TITLE
[a11y]: Tile [clickable with custom icon] - SR icon bug

### ISSUE THREAD

**Issue by andreancardona** [carbon-design-system/carbon#15268]
### Package

@carbon/react

### Browser

FF, Safari & iPhone

### Package version

@carbon/react@1.41.0

### Automated testing tool and ruleset

Manual QA

### Assistive technology

VO & JAWS

### Package

@carbon/react

### Browser

FF, Safari & iPhone

### Package version

@carbon/react@1.41.1

### Automated testing tool and ruleset

Manual QA

### Assistive technology

VO & JAWS

### Description
Screen reader is not announcing about the custom icon in the element

Screen reader is just announcing about the tile element and not about the custom icon in it.

### Recommended Fix
Screen reader should announce about the custom icon also along with the tile information.

People who uses screen reader will not be aware of the icon in the tile and the use of it.

### WCAG 2.1 Violation
4.1.2 Name, Role, Value

**Comment by andreancardona** [comment]
This issue works as expected within the browser. Closing for now.
